nautobot/ipam/filters.py

```diff
@@ -297,35 +319,49 @@ class PrefixFilterSet(
 
     class Meta:
         model = Prefix
-        fields = ["id", "is_pool"]
+        fields = ["id", "is_pool", "prefix"]
 
     def search(self, queryset, name, value):
-        if not value.strip():
+        value = value.strip()
+        if not value:
             return queryset
         qs_filter = Q(description__icontains=value)
         try:
-            prefix = str(netaddr.IPNetwork(value.strip()).cidr)
-            qs_filter |= Q(prefix__net_contains_or_equals=prefix)
+            query = netaddr.IPNetwork(value.strip())
+            qs_filter |= Q(
+                prefix_length__lte=query.prefixlen,
+                network__lte=bytes(query.network),
+                broadcast__gte=bytes(query.broadcast),
+            )
         except (AddrFormatError, ValueError):
             pass
         return queryset.filter(qs_filter)
 
     def filter_prefix(self, queryset, name, value):
-        if not value.strip():
+        value = value.strip()
+        if not value:
             return queryset
+        # filter for Prefix models equal to |value|
         try:
-            query = str(netaddr.IPNetwork(value).cidr)
-            return queryset.filter(prefix=query)
+            return queryset.net_equals(netaddr.IPNetwork(value))
         except (AddrFormatError, ValueError):
             return queryset.none()
 
+    def filter_prefix_length_eq(self, queryset, name, value):
+        return queryset.filter(prefix_length__exact=value)
+
+    def filter_prefix_length_lte(self, queryset, name, value):
+        return queryset.filter(prefix_length__lte=value)
+
+    def filter_prefix_length_gte(self, queryset, name, value):
+        return queryset.filter(prefix_length__gte=value)
+
```

Comment:
Are these not accounted for by default with an integer field type? I feel like they would be! Did you run into an issue that required these to be defined?

Reply:
I did yes, I can repro the error if you'd like